linera-core/src/updater.rs

```diff
@@ -278,13 +277,7 @@ where
 
         let mut missing_blobs = Vec::new();
         for blob_id in blob_ids {
-            if let Some(blob) = self
-                .local_node_recent_hashed_blobs
-                .lock()
-                .await
-                .get(blob_id)
-                .cloned()
-            {
+            if let Some(blob) = self.local_node_recent_hashed_blobs.get(blob_id).await {
```

Comment:
I think this could be refactored to only get the lock once, either using `subtract_cached_items_from` or the possibly new `try_get_many`.